tests/requirements.txt

```diff
@@ -4,29 +4,46 @@
-freezegun==0.3.7          # via -r requirements.in
-isort==4.2.5              # via pylint
-itsdangerous==0.24        # via flask
-jinja2==2.8               # via flask
-jsonschema==2.5.1         # via -r requirements.in
-lazy-object-proxy==1.2.2  # via astroid
-markupsafe==0.23          # via jinja2
-mccabe==0.5.2             # via pylint
-py==1.4.31                # via pytest
-pydantic==1.5.1           # via -r requirements.in
-pylint==1.6.4             # via -r requirements.in
-pytest-cov==2.5.1         # via -r requirements.in
-pytest-flask==0.10.0      # via -r requirements.in
-pytest==3.0.3             # via -r requirements.in, pytest-cov, pytest-flask
-python-dateutil==2.5.3    # via freezegun
-requests==2.11.1          # via -r requirements.in
-semantic-version==2.8.4   # via -r requirements.in
-six==1.10.0               # via astroid, freezegun, pylint, python-dateutil, tenacity
+astroid==2.4.2            # via pylint
+attrs==20.3.0             # via jsonschema, pytest
+certifi==2020.11.8        # via requests
+chardet==3.0.4            # via requests
+click==7.1.2              # via flask
+coverage==5.3             # via pytest-cov
+flask==1.1.2              # via pytest-flask
+flexmock==0.10.4          # via -r requirements.in
+freezegun==1.0.0          # via -r requirements.in
+idna==2.10                # via requests
+importlib-metadata==2.0.0  # via jsonschema, pluggy, pytest
+iniconfig==1.1.1          # via pytest
+isort==5.6.4              # via pylint
+itsdangerous==1.1.0       # via flask
+jinja2==2.11.2            # via flask
+jsonschema==3.2.0         # via -r requirements.in
+lazy-object-proxy==1.4.3  # via astroid
+markupsafe==1.1.1         # via jinja2
+mccabe==0.6.1             # via pylint
+packaging==20.4           # via pytest
+pluggy==0.13.1            # via pytest
+py==1.9.0                 # via pytest
+pydantic==1.7.2           # via -r requirements.in
+pylint==2.6.0             # via -r requirements.in
+pyparsing==2.4.7          # via packaging
+pyrsistent==0.17.3        # via jsonschema
+pytest-cov==2.10.1        # via -r requirements.in
+pytest-flask==1.1.0       # via -r requirements.in
+pytest==6.1.2             # via -r requirements.in, pytest-cov, pytest-flask
+python-dateutil==2.8.1    # via freezegun
+requests==2.25.0          # via -r requirements.in
+semantic-version==2.8.5   # via -r requirements.in
+six==1.15.0               # via astroid, jsonschema, packaging, python-dateutil, tenacity
 tenacity==6.2.0           # via -r requirements.in
-werkzeug==0.11.11         # via flask, pytest-flask
-wrapt==1.10.8             # via astroid
+toml==0.10.2              # via pylint, pytest
+typed-ast==1.4.1          # via astroid
+urllib3==1.26.2           # via requests
+werkzeug==1.0.1           # via flask, pytest-flask
+wrapt==1.12.1             # via astroid
+zipp==3.4.0               # via importlib-metadata
+
+# The following packages are considered to be unsafe in a requirements file:
+# setuptools
```

Comment:
Please make sure while generating this file, your virtual env is activated and it has no dependencies installed previously. 
Then ideally the process should be
1. Create and Activate new Virtual Env
2. Add new Deps that are lying in Dockerfile or so in requirements.in
3. `pip-compile --output-file=requirements.txt requirements.in`
4. `pip install -r requirements.txt` to check any Errors.

Reply:
I am checking directly on the cluster after building the image.Is that not okay?